artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/AddressControlImpl.java

```diff
@@ -373,6 +372,86 @@ public String sendMessage(final Map<String, String> headers,
       return MBeanInfoHelper.getMBeanAttributesInfo(AddressControl.class);
    }
 
+   @Override
+   public void pause() throws Exception {
+      pause(false);
+   }
+
+   @Override
+   public void pause(boolean persist) throws Exception {
+      if (AuditLogger.isEnabled()) {
+         AuditLogger.pause(addressInfo);
+      }
+      checkStarted();
+
+      clearIO();
+      try {
+         if (!addressInfo.isPaused()) {
+            addressInfo.pause();
+            if (persist) {
+               server.getStorageManager().storeAddressStatus(addressInfo.getId(), AddressStatus.PAUSED);
+            }
+            Bindings bindings = server.getPostOffice().lookupBindingsForAddress(addressInfo.getName());
+            if (bindings != null) {
+               for (Binding binding : bindings.getBindings()) {
+                  if (binding instanceof QueueBinding) {
+                     QueueControl coreQueueControl = (QueueControl) managementService.getResource(ResourceNames.QUEUE + binding.getUniqueName().toString());
```

Comment:
Instead of going through the `managementService` you can just call `binding.getQueue().pause(boolean)`. Be sure you pass the `persist` option through to the `Queue`.

Reply:
ok